src/modules/dashboard/components/EditDomainDialog/EditDomainDialogForm.tsx

```diff
@@ -0,0 +1,217 @@
+const EditDomainDialogForm = ({
+  back,
+  colony,
+  colony: { domains },
+  handleSubmit,
+  isSubmitting,
+  isValid,
+  setFieldValue,
+  values: { domainId },
+}: Props & FormikProps<FormValues>) => {
+  const [domainColor, setDomainColor] = useState(Color.LightPink);
+
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canEditDomain = hasRegisteredProfile && canArchitect(allUserRoles);
+
+  const domainOptions = useMemo(
+    () =>
+      sortBy(
+        domains.map(({ name, ethDomainId }) => ({
+          value: ethDomainId.toString(),
+          label: name,
+        })),
+        ['value'],
+      ),
+
+    [domains],
+  );
+
+  const handleDomainChange = (selectedDomainId) => {
+    const selectedDomain = domains.find(
+      (domain) => domain.ethDomainId.toString() === selectedDomainId,
+    );
+    if (selectedDomain) {
+      setFieldValue(
+        'domainId',
+        selectedDomain.ethDomainId.toString() || undefined,
+      );
+      setFieldValue('domainColor', selectedDomain.color || undefined);
+      setFieldValue('domainName', selectedDomain.name || undefined);
+      setFieldValue('domainPurpose', selectedDomain.description || undefined);
+      setDomainColor(selectedDomain.color);
+    }
+  };
+  useEffect(() => {
+    if (domainId) {
+      handleDomainChange(domainId);
+    }
+  }, []);
+
+  return (
+    <>
+      <DialogSection appearance={{ theme: 'heading' }}>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={MSG.titleEdit}
+          className={styles.title}
+        />
+      </DialogSection>
+      {!canEditDomain && (
+        <DialogSection>
+          <PermissionRequiredInfo requiredRoles={[ColonyRole.Architecture]} />
+        </DialogSection>
+      )}
+      <DialogSection>
+        <div className={styles.nameAndColorContainer}>
+          <div className={styles.domainName}>
+            <Select
+              options={domainOptions}
+              label={MSG.domain}
+              onChange={handleDomainChange}
+              name="domainId"
+              appearance={{ theme: 'grey', width: 'fluid' }}
+              disabled={!canEditDomain}
+            />
+          </div>
+          <ColorSelect
+            activeOption={domainColor}
+            appearance={{ alignOptions: 'right' }}
+            onColorChange={setDomainColor}
+            disabled={!canEditDomain}
+            name="domainColor"
+          />
+        </div>
+      </DialogSection>
+      <DialogSection>
+        <div className={styles.nameAndColorContainer}>
+          <div className={styles.domainName}>
+            <Input
+              label={MSG.name}
+              name="domainName"
+              appearance={{ colorSchema: 'grey', theme: 'fat' }}
+              disabled={!canEditDomain}
+              maxLength={20}
```

Comment:
The team name input should the same length as the others

![Screenshot from 2021-02-03 00-15-51](https://user-images.githubusercontent.com/1193222/106669924-132af380-65b5-11eb-9ebd-a388e730ee56.png)

Currently it's a bit shorter